### PR TITLE
Add debug log to acquiring a connection

### PR DESCRIPTION
**Pull request by akc42**
Useful extra information for debugging connection useage

### CODE REVIEW

File: lib/connection-pool.js [hunk 1]
```diff
@@ -220,6 +221,7 @@ function setFree(pooled) {
     pooled.status = FREE;
     pooled.timeout = setTimeout(function() {
         self.log('closing idle connection: ' + pooled.id);
+        pooled.status = PENDING;
```
Comment (ben-page) [inline lib/connection-pool.js]: This isn't necessary. The `end` event handler is called synchronously in the `close()` function. The connection status will be set to RETRY by that `end` event handler before any other runs.

**Comment by ben-page** [comment]
Thanks for this. I think your extra log information is a good contribution. However, the other change is not necessary. If you remove it, I'll pull the PR.

**Comment by ben-page** [comment]
Since this was two commits, I just cherry picked the one that I wanted to include. Thanks for your contribution.

**Comment by akc42** [comment]
Yea.  Sorry I am snowed under with real life and have not only forgotten how to make a proper one issue PR, but have other more pressing stuff.  I was trying to schedule to do it, but I am glad you've just gone ahead anyway..

**Pull request closed without merging**